common/platform.h

```diff
@@ -37,56 +37,116 @@ namespace Common {
+#define GET_ENUM_VAL(name) GET_ENUM_VAL_EXPAND((name))
+#else
+#define GET_ENUM_VAL_IMPL(val, hex) val
+#define GET_ENUM_VAL(name) GET_ENUM_VAL_IMPL(name)
+#endif
+
+// List of platforms values as int and string literals, to be used in platform enum and gui options
+#define kPlatformApple2GS_VAL 0x40, "\x40"
+#define kPlatformApple2_VAL 0x41, "\x41"
+#define kPlatform3DO_VAL 0x42, "\x42"
+#define kPlatformAcorn_VAL 0x43, "\x43"
+#define kPlatformAmiga_VAL 0x44, "\x44"
+#define kPlatformAtari8Bit_VAL 0x45, "\x45"
+#define kPlatformAtariST_VAL 0x46, "\x46"
+#define kPlatformC64_VAL 0x47, "\x47"
+#define kPlatformAmstradCPC_VAL 0x48, "\x48"
+#define kPlatformDOS_VAL 0x49, "\x49"
+#define kPlatformPC98_VAL 0x4A, "\x4A"
+#define kPlatformWii_VAL 0x4B, "\x4B"
+#define kPlatformCoCo_VAL 0x4C, "\x4C"
+#define kPlatformCoCo3_VAL 0x4D, "\x4D"
+#define kPlatformFMTowns_VAL 0x4E, "\x4E"
+#define kPlatformLinux_VAL 0x4F, "\x4F"
+#define kPlatformMacintosh_VAL 0x50, "\x50"
+#define kPlatformPCEngine_VAL 0x51, "\x51"
+#define kPlatformNES_VAL 0x52, "\x52"
+#define kPlatformSegaCD_VAL 0x53, "\x53"
+#define kPlatformWindows_VAL 0x54, "\x54"
+#define kPlatformPSX_VAL 0x55, "\x55"
+#define kPlatformPS2_VAL 0x56, "\x56"
+#define kPlatformPS3_VAL 0x57, "\x57"
+#define kPlatformXbox_VAL 0x58, "\x58"
+#define kPlatformCDi_VAL 0x59, "\x59"
+#define kPlatformIOS_VAL 0x5A, "\x5A"
+#define kPlatformAndroid_VAL 0x5B, "\x5B"
+#define kPlatformOS2_VAL 0x5C, "\x5C"
+#define kPlatformBeOS_VAL 0x5D, "\x5D"
+#define kPlatformPocketPC_VAL 0x5E, "\x5E"
+#define kPlatformMegaDrive_VAL 0x5F, "\x5F"
+#define kPlatformSaturn_VAL 0x60, "\x60"
+#define kPlatformPippin_VAL 0x61, "\x61"
+#define kPlatformMacintoshII_VAL 0x62, "\x62"
+#define kPlatformShockwave_VAL 0x63, "\x63"
+#define kPlatformZX_VAL 0x64, "\x64"
+#define kPlatformTI994_VAL 0x65, "\x65"
+#define kPlatformNintendoSwitch_VAL 0x66, "\x66"
+
 /**
  * List of game platforms. Specifying a platform for a target can be used to
  * give the game engines a hint for which platform the game data file are.
  * This may be optional or required, depending on the game engine and the
  * game in question.
  */
 enum Platform : int8 {
-	kPlatformDOS,
-	kPlatformAmiga,
-	kPlatformAmstradCPC,
-	kPlatformAtari8Bit,
-	kPlatformAtariST,
-	kPlatformMacintosh,
-	kPlatformFMTowns,
-	kPlatformWindows,
-	kPlatformNES,
-	kPlatformC64,
-	kPlatformCoCo,
-	kPlatformCoCo3,
-	kPlatformLinux,
-	kPlatformAcorn,
-	kPlatformSegaCD,
-	kPlatform3DO,
-	kPlatformPCEngine,
-	kPlatformApple2,
-	kPlatformApple2GS,
-	kPlatformPC98,
-	kPlatformWii,
-	kPlatformPSX,
-	kPlatformPS2,
-	kPlatformPS3,
-	kPlatformXbox,
-	kPlatformCDi,
-	kPlatformIOS,
-	kPlatformAndroid,
-	kPlatformOS2,
-	kPlatformBeOS,
-	kPlatformPocketPC,
-	kPlatformMegaDrive,
-	kPlatformSaturn,
-	kPlatformPippin,
-	kPlatformMacintoshII,
-	kPlatformShockwave,
-	kPlatformZX,
-	kPlatformTI994,
-	kPlatformNintendoSwitch,
-
-	kPlatformUnknown = -1
+    kPlatformDOS = GET_ENUM_VAL(kPlatformDOS_VAL),
```

Comment:
This approach still requires two lists to sync between them. Can we somehow have only one list? Maybe you need to move the defines to `platform_intern.h` and include it twice?